airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/manifest.yaml

```diff
@@ -0,0 +1,1404 @@
+              - object
+  - name: call_legs
+    type: DeclarativeStream
+    retriever:
+      type: SimpleRetriever
+      paginator:
+        type: DefaultPaginator
+        page_token_option:
+          type: RequestPath
+        pagination_strategy:
+          type: CursorPagination
+          cursor_value: '{{ response.get("next_page", {}) }}'
+          stop_condition: '{{ response.get("count", {}) <= 1 }}'
+      ignore_stream_slicer_parameters_on_paginated_requests: true
+      requester:
+        path: /stats/incremental/legs
+        type: HttpRequester
+        url_base: "https://{{ config['subdomain'] }}.zendesk.com/api/v2/channels/voice/"
+        http_method: GET
+        authenticator: "#/definitions/authenticator"
+        request_headers: {}
+        request_body_json: {}
+        request_parameters: {}
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - legs
+            - "*"
+      partition_router: []
+    primary_key:
+      - id
+    schema_loader:
+      type: InlineSchemaLoader
+      schema:
+        type: object
+        $schema: "http://json-schema.org/schema#"
+        properties:
+          id:
+            type:
+              - "null"
+              - integer
+          type:
+            type:
+              - "null"
+              - string
+          call_id:
+            type:
+              - "null"
+              - integer
+          user_id:
+            type:
+              - "null"
+              - integer
+          agent_id:
+            type:
+              - "null"
+              - integer
+          duration:
+            type:
+              - "null"
+              - integer
+          hold_time:
+            type:
+              - "null"
+              - integer
+          talk_time:
+            type:
+              - "null"
+              - integer
+          created_at:
+            type:
+              - "null"
+              - string
+          updated_at:
+            type:
+              - "null"
+              - string
+            format: "date-time"
+          call_charge:
+            type:
+              - "null"
+              - string
+          wrap_up_time:
+            type:
+              - "null"
+              - integer
+          available_via:
+            type:
+              - "null"
+              - string
+          minutes_billed:
+            type:
+              - "null"
+              - integer
+          quality_issues:
+            type:
+              - "null"
+              - array
+          completion_status:
+            type:
+              - "null"
+              - string
+    incremental_sync:
+      type: DatetimeBasedCursor
+      cursor_field: "{{ config['start_date'] }}"
+      start_datetime:
+        type: MinMaxDatetime
+        datetime: "{{ config['start_date'] }}"
+        datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+      datetime_format: "%s"
```

Comment:
🤔  The datetime string is the format returned by the API for each record, while the epoch seconds is the format required for the `start_time` param in requests. I didn't specifically test anything around emitted state message format, though, any tips on how to do that @maxi297?

Reply:
This would make sense. I'm just a bit worried because our state could have a cursor value of "1709741421" or "2021-01-01T00:00:00Z" depending on where the value comes from. I would test read with an incremental stream with both those values within the state juste to make sure or check if there is a unit test in DatetimeBasedCursor that validates that both inbound and outbound format can be used for the state

Reply:
> check if there is a unit test in DatetimeBasedCursor that validates that both inbound and outbound format can be used for the state

Ah, I was just working with those exact unit tests as part of https://github.com/airbytehq/airbyte/pull/35843: there are unit tests verifying `DatetimeBasedCursor` can handle varying formats, and the time strings get parsed into `datetime` objects for comparison, so changing formats shouldn't affect the logical ordering.

Reply:
It still seems like this would be a breaking change. The previous version of the connector needs to support the new format if we want to be able to revert.

Reply:
Do you consider this a blocker? I.e. do you need to see a PR updating the legacy source definition to handle unix epoch seconds as well as datetime strings merged to `master` before this can be merged?

Reply:
Actually, looking more closely at this, I do not think that this would be particularly likely to break anyone's syncs in practice: the updated definition does not define an `end_datetime` for incremental sync, so the record's cursor value—the `updated_at` datetime string, as before—would be used for the state if even a single record were emitted in the sync. I'm not sure that taking the time to avoid this edge case for a connector that only 12 cloud workspaces use would really be striking the best balance between careful work and development speed.

Reply:
I think the question is: can we run a sync on version `0.1.13` with a state the the low-code source generated? If not, this is a breaking change. I'm fine with just documenting the breaking change if it is hard to avoid

Reply:
I manually attempted such a sync, and it triggered a python exception from trying to parse an epoch timestamp with pendulum's datestring code; the breaking change is now documented.